### PR TITLE
Exposing the source solidity files on the NPM package

### PR DESCRIPTION
**Pull request by brickpop**
## Description

Please include a summary of the change and be sure you follow the contributions rules we do provide [here](./CONTRIBUTIONS.md)

Task: APP-1800

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

**Comment by Michael-A-Heuer** [comment]
`&& cp -a ./src/*`

Does this mean we would import files as 
```js
import '@aragon/core-contracts/core/plugin/Plugin.sol';
import '@aragon/core-contracts/framework/plugin/setup/PluginSetup.sol';
```
or, after we changed the paths, as 
```js
import '@aragon/os-contracts/core/plugin/Plugin.sol';
import '@aragon/os-contracts/framework/plugin/setup/PluginSetup.sol';
```
?

That would be the best way IMO.

**Comment by brickpop** [comment]
@Michael-A-Heuer yes, the files being published are under `dist` and these now include the solidity `src/*` files, in addition to just the TS files.

You can run `yarn build && yarn build:npm && npm pack`  and then install this `tar.gz` file elsewhere and see if you can import

File: packages/contracts/package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
   "scripts": {
     "test": "hardhat test",
     "build": "hardhat compile",
-    "build:npm": "rollup --config rollup.config.ts",
+    "build:npm": "rollup --config rollup.config.ts && cp -a ./src/* ./dist",
```
Comment (Michael-A-Heuer) [inline packages/contracts/package.json:17]: The resulting import path would be 

import "@aragon/core-contracts/**dist**/core/dao/DAO.sol";

I would do it like OZ, ENS, and GSN and directly put it in the top-level folder.

import "@aragon/core-contracts/core/dao/DAO.sol";

Moreover, I would only be more selective with the folders:

Include
- `core` 
- `framework`
- `utils`

Potentially include
- `plugins`
- `token`

Exclude
- `test`

Comment (brickpop) [reply packages/contracts/package.json:17]: The problem with this is that this forces us to copy the source folders to the root of the `contract` folder, which is very cumbersome

Comment (brickpop) [reply packages/contracts/package.json:17]: We could also whitelist `dist` and `src` and then imports would look like this
Alternatively, having a kind of "chroot" within `dist`.  Copying the package.json file there and doing the publish from there

Comment (Michael-A-Heuer) [inline packages/contracts/package.json:17]:
```suggestion
    "build:npm": "rollup --config rollup.config.ts && cp -a ./src/core ./ && cp -a ./src/framework . && cp -a ./src/utils . &&  cp -a ./src/plugins . && cp -a ./src/token .",
```
Wouldn't this work?

Comment (brickpop) [reply packages/contracts/package.json:17]: the package.json and/or `.npmignore` files need to be updated for this

And this script line will have to be updated if we ever add another folder

Comment (Michael-A-Heuer) [reply packages/contracts/package.json:17]: I would do it to keep the import paths as short as possible.
Including `dist` for no reason is annoying/ugly.

Comment (novaknole) [reply packages/contracts/package.json:17]: I agree. `dist` shouldn't be in the path and for the first time, let's include plugins and token. And ofc exclude `test`.

**Comment by mathewmeconry** [comment]
we go a different route. I'll create another pull as soon as the changes are ready

**Pull request closed without merging**